src/ClientFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```

Comment:
What should the default be? `8.8.8.8`?

Actually, I don't really know why you have to select a DNS resolver manually. I'm not too familiar with how React works, but I can look into it.